src/gui/src/state/index.ts

```diff
@@ -53,6 +63,33 @@ export const useGraphStore = create<Action & State>(set => {
     updateStamp: (stamp: string) => set(() => ({ stamp })),
     updateTheme: (theme: State['theme']) => set(() => ({ theme })),
     reset: () => set(initialState),
+    updateSavedProjects: (savedProjects: State['savedProjects']) =>
+      set(() => ({ savedProjects })),
+    saveProject: (item: DashboardDataProject) => {
+      const savedProjects = get().savedProjects ?? []
+      let updatedProjects = [...savedProjects]
+      const isProjectSaved = updatedProjects.find(
+        savedProject => savedProject.name === item.name,
+      )
+
+      if (!isProjectSaved) {
+        updatedProjects.push(item)
+      } else {
+        updatedProjects = updatedProjects.filter(
+          savedProject => savedProject.name !== item.name,
```

Comment:
same as above, we need to use `item.path` anywhere that we check for a unique project